README.md

```diff
@@ -1,41 +1,150 @@
+Property values should:
+- be a boolean, number, string, or enum (of strings) type.
+- strings: be consistent within all possible values, opting for either `Proper Case`, `kebab-case`,  or `snake-case`. The default selection should be `Proper Case`.
 
-The top level files, such as [primitives](./src//primitives.ts) contain generic data that is supposed to be reused by different applications. This data describes the events being logged, such as `EventName.MENU_CLICK`, which reflects an event on a menu. Specific event context should be captured in additional fields beyond the event name to create user-friendly hierarchy groupings.
+## Defining Events In Code
 
-## Example Apps
+Event names, property names, and property values should all be defined in Javascript using `enum`'s and follow the following conventions:
+- `enum` names should be as specific as possible (e.g `DocsProtocolVersion` over `Version`).
+- `enum` names should be in `PascalCase`.
+- Code values of enums should be named in `UPPER_SNAKE_CASE`.
 
-This package is used into a few of Uniswap Labs repos:
+## Organizing Events
 
-- [Interface](https://github.com/Uniswap/widgets-demo/tree/nextjs)
-- [Docs](https://github.com/Uniswap/docs)
+These enumerations and events can be shared across products or be specific to a product or use case. All event names, property names, and property values should be defined using enumerations placed in the best folder for ease of reference, and can be split out into separate files when needed. When defining enumerations, consider that the package exports all names at the top level, so adding a specific prefix for your product or use case may be useful. If your use cases are highly generalizable, you can define your enumeration as a primitive or reuse an existing primitive.
 
+Top level files are for such as [primitives](./src/primitives.ts) contain generic data that is supposed to be reused by different applications. This data describes the events being logged, such as `EventName.MENU_CLICK`, which reflects an event on a menu. Specific event context should be captured in additional fields beyond the event name to create user-friendly hierarchy groupings.
+
+Folders should be made by origin or product. For example, [docs](./src/docs) contains events specific to our documentation. Folders can be grouped by product when multiple origins could use these events, such as general actions like swapping.
+
+## Testing Events
+
+For rapid development, a convenience flow is available that allows you to do the following in a single command:
+- create a tarball copy of the latest events
+- copy this tarball to the specified project
+- install this package in the specified project
+
+To set this up, define the following environment variables (in your bash/zsh profile or as desired):
```

Comment:
It could be super helpful here to add an example with a sample file directory

Reply:
Does that add anything? In this case it would just be a path specific to someone's computer that doesn't work for anyone copy/pasting

Reply:
Perhaps not useful for copying/pasting but valuable for one's understanding. I imagine a junior dev trying to go through this. Regardless, if we don't think it adds value, we could skip.